src/Storages/MergeTree/GinIndexStore.h

```diff
@@ -185,6 +188,9 @@ class GinIndexStore
 
     std::mutex mutex;
 
+    /// Not thread-safe, protected by mutex
+    std::optional<UInt32> next_available_segment_id = std::nullopt;
```

Comment:
It seems that making `next_available_segment_id` an `std::optional` complicates things. What if we change it to a `UInt32` with default-initialization = 1? As far as I see, the previous implementation _always_ wrote the segment id file (even if there was just one segment id).